lib/useOnyx.ts

```diff
@@ -232,8 +238,8 @@ function useOnyx<TKey extends OnyxKey, TReturnValue = OnyxValue<TKey>>(key: TKey
 
         // If the previously cached value is different from the new value, we update both cached value
         // and the result to be returned by the hook.
-        // If the cache was set for the first time, we also update the cached value and the result.
-        const isCacheSetFirstTime = previousValueRef.current === null && hasCacheForKey;
+        // If the cache was set for the first time or we have a pending Onyx.clear() task, we also update the cached value and the result.
+        const isCacheSetFirstTime = previousValueRef.current === null && (hasCacheForKey || OnyxCache.hasPendingTask(TASK.CLEAR));
```

Comment:
What does it mean for the cache to be "set first time"? Like, as in the first time the hook runs? It makes sense to consider it "set" when it has cache value. But, can we improve the documentation to explain why we should consider the cache "set" if there is a pending clear task?

Reply:
I refactored the comments, please let me know if it's clearer now.